Occasional panic writing to websocket

I saw this one other time, but wasn't setup to capture the log.  This error occasionally happens and will stop the server / automation from continuing.

```
2024/05/06 07:38:14 Received command interaction.pressKeys
panic: concurrent write to websocket connection

goroutine 19 [running]:
github.com/gorilla/websocket.(*messageWriter).flushFrame(0xc000461c50, 0x1, {0xc000426c7b?, 0x10?, 0x60bde0?})
        C:/Users/gnarf/go/pkg/mod/github.com/gorilla/websocket@v1.5.0/conn.go:617 +0x4b8
github.com/gorilla/websocket.(*Conn).WriteMessage(0xc0001542c0, 0x22?, {0xc000426c00, 0x7b, 0x80})
        C:/Users/gnarf/go/pkg/mod/github.com/gorilla/websocket@v1.5.0/conn.go:770 +0x127
Server/server.(*Server).sendMessage(...)
        F:/Code/aria-at/nvda-at-automation/Server/server/server.go:70
Server/server.(*Server).handleNewSessionCommand.func2.1({0xc00009c1f0?, 0xc0004f3010?, 0xc00009c350?})
        F:/Code/aria-at/nvda-at-automation/Server/server/server.go:114 +0x91
Server/sse.(*Parser).tryDispatchEvent(0xc000461f98)
        F:/Code/aria-at/nvda-at-automation/Server/sse/eventstream.go:69 +0x42
Server/sse.(*Parser).Process(0xc0004f2aa8?, {0xc0004f2aa8, 0x7})
        F:/Code/aria-at/nvda-at-automation/Server/sse/eventstream.go:58 +0x1a5
Server/client.(*NVDA).RegisterOnLineCallback(0xc00004a4c0, 0xc000461fa8)
        F:/Code/aria-at/nvda-at-automation/Server/client/client.go:253 +0x4e5
Server/server.(*Server).handleNewSessionCommand.func2()
        F:/Code/aria-at/nvda-at-automation/Server/server/server.go:121 +0x7e
created by Server/server.(*Server).handleNewSessionCommand in goroutine 18
        F:/Code/aria-at/nvda-at-automation/Server/server/server.go:124 +0x291
```

--- a/Server/server/server.go
+++ b/Server/server/server.go
@@ -11,9 +11,11 @@ import (
 	"github.com/gorilla/websocket"
 	"log"
 	"net/http"
+	"sync"
 )
 
 type Server struct {
+	mutex sync.Mutex
 	connection *websocket.Conn
 	sessionID  *string
 	client     *client.NVDA
@@ -59,7 +61,7 @@ func (s *Server) serve(w http.ResponseWriter, r *http.Request) {
 
 		res := s.handleMessage(message)
 
-		err = c.WriteMessage(websocket.TextMessage, res)
+		err = s.sendMessage(res)
 		if err != nil {
 			log.Println("send:", err)
 		}
@@ -67,6 +69,8 @@ func (s *Server) serve(w http.ResponseWriter, r *http.Request) {
 }
 
 func (s *Server) sendMessage(message []byte) error {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
 	return s.connection.WriteMessage(websocket.TextMessage, message)
 }
 

